agent/src/main/resources/build-scan-init.gradle

```diff
@@ -201,6 +201,20 @@ static boolean isNotAtLeast(String versionUnderTest, String referenceVersion) {
     GradleVersion.version(versionUnderTest) < GradleVersion.version(referenceVersion)
 }
 
+void applyPluginExternally(PluginManager pluginManager, String pluginClassName) {
+    def oldValue = System.getProperty('gradle.enterprise.externally-applied')
+    System.setProperty('gradle.enterprise.externally-applied', 'true')
```

Comment:
I suggest moving the string `gradle.enterprise.externally-applied` into a local variable at the top of the method so avoid the 4-time repetition of the literal.

Reply:
Done in bc54aa2ec5ff010cf61face1d0d8ac6a45c2ddb7